crates/rolldown_plugin/src/plugin_driver/mod.rs

```diff
@@ -21,23 +27,25 @@ impl PluginDriver {
     file_emitter: &SharedFileEmitter,
   ) -> SharedPluginDriver {
     Arc::new_cyclic(|plugin_driver| {
-      let with_context = plugins
-        .into_iter()
-        .map(|plugin| {
-          (
-            plugin,
-            PluginContext {
-              plugin_driver: Weak::clone(plugin_driver),
-              resolver: Arc::clone(resolver),
-              file_emitter: Arc::clone(file_emitter),
-              module_table: None,
-            }
-            .into(),
-          )
-        })
-        .collect::<Vec<_>>();
+      let mut index_plugins = IndexPluginable::with_capacity(plugins.len());
+      let mut index_contexts = IndexPluginContext::with_capacity(plugins.len());
 
-      Self { plugins: with_context }
+      plugins.into_iter().for_each(|plugin| {
+        let plugin_idx = index_plugins.push(Arc::clone(&plugin));
+        index_contexts.push(
+          PluginContext {
+            skipped_resolve_calls: vec![],
```

Comment:
Looks like `context.skipped_resolve_calls` is always empty. Is it useful or could be removed? @underfin cc

Reply:
It will have value if has skip-self resolve calls.